packages/react/src/authentication/hooks/useSocialLogin.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  type Config,
+  type PostAccountLinkData,
+  type PostSocialLoginData,
+} from '@farfetch/blackout-client';
+import {
+  createAccountLink as createAccountLinkAction,
+  socialLogin as socialLoginAction,
+} from '@farfetch/blackout-redux';
+import { useCallback } from 'react';
+import useAction from '../../helpers/useAction.js';
+
+function useSocialLogin() {
+  const socialLogin = useAction(socialLoginAction);
+  const createAccountLink = useAction(createAccountLinkAction);
+  const login = useCallback(
+    async (data: PostSocialLoginData, config?: Config) => {
+      if (!data) {
+        return Promise.reject(new Error('No data was specified.'));
+      }
+
+      return await socialLogin(data, config);
+    },
+    [socialLogin],
+  );
+
+  const accountLink = useCallback(
+    async (data: PostAccountLinkData, config?: Config) => {
+      if (!data) {
+        return Promise.reject(new Error('No data was specified.'));
+      }
+
+      return await createAccountLink(data, config);
+    },
+    [createAccountLink],
+  );
+
+  return {
+    actions: {
+      login,
+      accountLink,
```

Comment:
I would change name of the function to better indicate the action to be performed, `createAccountLink` is more indicative than simply `accountLink`, just a suggestion if you disagree you can close this thread.